src/base/base_components.jl

```diff
@@ -0,0 +1,143 @@
+
+@component function MaterialStream(ms::MaterialSource;phase="unknown",name)
+    @named c1 = MaterialConnector(ms)
+    @named c2 = MaterialConnector(ms)
+    mcons = [c1,c2]
+    
+    vars = @variables begin
+        T(t),               [description="temperature"]         #, unit=u"K"]
+        ϱ(t),               [description="density"]             #, unit=u"mol m^-3"]
+        p(t),               [description="pressure"]            #, unit=u"Pa"]
+        h(t),               [description="molar enthalpy"]      #, unit=u"J mol^-1"]
+        n(t),               [description="molar flow "]         #, unit=u"mol s^-1"]
+        xᵢ(t)[1:ms.N_c],    [description="mole fractions"]      #, unit=u"mol mol^-1"]
+    end
+
+    eqs = [
+        # EOS 
+        ϱ ~ ms.molar_density(p,T,xᵢ;phase=phase),
+        h ~ ms.VT_enthalpy(ϱ,T,xᵢ),
+        1.0 ~ sum(collect(xᵢ)),
+        # Connector "1"
+        T ~ c1.T,
+        ϱ ~ c1.ϱ,
+        p ~ c1.p,
+        h ~ c1.h,
+        n ~ c1.n,
+        scalarize(xᵢ .~ c1.xᵢ)...,
+        # Connector "2"
+        T ~ c2.T,
+        ϱ ~ c2.ϱ,
+        p ~ c2.p,
+        h ~ c2.h,
+        n ~ -c2.n,
+        scalarize(xᵢ .~ c2.xᵢ)...,   
+    ]
+
+    return ODESystem(eqs, t, collect(Iterators.flatten(vars)), []; name, systems=mcons)
+end
+
+@connector function MaterialConnector(ms::MaterialSource;name)
+    vars = @variables begin
+        T(t),               [description="temperature", output=true]        #, unit=u"K"]
+        ϱ(t),               [description="density", output=true]            #, unit=u"mol m^-3"]
+        p(t),               [description="pressure"]                        #, unit=u"Pa"]
+        h(t),               [description="molar enthalpy", output=true]     #, unit=u"J mol^-1"]
+        xᵢ(t)[1:ms.N_c],    [description="mole fractions", output=true]     #, unit=u"mol mol^-1"]
+        n(t),               [description="total molar flow", connect=Flow]  #, unit=u"mol s^-1"]
```

Comment:
What happens if you have 2 reactors connected into a tank, are `mole fractions` handled appropriately by the `connect` equations? I see from the commits that this went from Stream to output=true. Is `output=true` similar to Stream?

Reply:
Right, `Stream` connectors would be the better choice here, but their equations were missing as reported [here](https://discourse.julialang.org/t/stream-connections-of-subsystems/121279/1). The `output=true` is only used to prevent warnings when creating the system.

Reply:
I also encountered a very similar bug (singly-connected system works when I use an Across connector, fails when I use a Stream connector). We should open an MTK issue.

Reply:
I confirmed that the stream equations are missing in my case too. Haven't tried to get rid of the subsystem. I'll do an MWE on Friday if no one beats me to it.